### PR TITLE
Fix the hypershift-mce-kubevirt-conformance workflow

### PR DESCRIPTION
**Pull request by orenc1**
Also, `MonitorTest` suddenly started to fail for hypershift/kubevirt, also without running the live migration test.
Until we resolve the issue, we're ignoring failures originated from the MonitorTest, both for MCE and the dev conformance lanes.

### CODE REVIEW

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Review by davidvossel — CHANGES_REQUESTED** [review]
changing the cluster namespace here impacts the other MCE scripts. `local-cluster` is the expected default namespace for HCP when mce is in use. 

look at the output of this search in the release directory to see how prevalent the usage of local-cluster is

```
grep -r -n "local-cluster" ci-operator/step-registry/hypershift/mce/*
```

I think it would be more consistent if we could get the conformance tests to work accurately with whatever namespace is in use. Maybe the cluster namespace can be detected in `ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml` by using the CLUSTER_NAME... so maybe do something like `oc get hcp -A --no-headers | grep $CLUSTER_NAME` | awk '{ print $1 }'` or detect the namespace in some similar way

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance pull-ci-openshift-hypershift-main-e2e-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance pull-ci-openshift-hypershift-main-e2e-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

File: ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml
Comment (orenc1) [reply ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml]: yes, it's better because it's shorter and doesn't use additional binary (`jq`).
updated.

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve

looks good, lets see a few rehearsal's pass before merging this

**Comment by orenc1** [comment]
> /pj-rehearse hypershift-main-e2e-conformance

/pj-rehearse pull-ci-openshift-hypershift-main-e2e-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance pull-ci-openshift-hypershift-main-e2e-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance pull-ci-openshift-hypershift-main-e2e-conformance

**Comment by orenc1** [comment]
pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by davidvossel** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance pull-ci-openshift-hypershift-main-e2e-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by davidvossel** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance pull-ci-openshift-hypershift-main-e2e-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

File: ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml
Comment (davidvossel) [inline ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml:105]: error ->  `/bin/bash: -c: line 100: conditional binary operator expected`

this line should be

```
if [[ -n "${SKIP_MONITOR_TEST}" ]] && [[ $(grep "failed due to a MonitorTest failure" /tmp/openshift-tests.log) ]]; 
then
```

File: ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml
Comment (davidvossel) [inline ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml:106]: also, can you print here something like

`echo "overriding Monitor Test Failure" ` so we'll have something that indicates clearly when thsi occurs.

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

File: ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml
Comment (davidvossel) [inline ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml:100]: After looking at the logs, I still see failures but the conformance test result is success. I think we're getting the exit code of the `tee` command and not the `openshift-tests` command

look at the example below. The first exit code is 1 and the second is 0 despite the two greps failing in both examples. 

```
#!/bin/bash
#
echo "test" | grep "madeup"
echo $?

echo "test" | grep "madeup" | tee /tmp/somefile
echo $?
```

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-mce-conformance periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-kubevirt-conformance

File: ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml
Comment (orenc1) [reply ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml]: i'm now retrieving the exit code from `${PIPESTATUS[0]}`, which is accurately the exit code of `openshift-tests`, not `tee`.

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve
/hold

Lets see how the test results turn out. Even if we have a failure, i think this PR gets us closer and is good to merge. I just want to make sure we don't impact any of the other non kubevirt lanes

**Comment by davidvossel** [comment]
/pj-rehearse  pull-ci-openshift-hypershift-main-e2e-conformance

**Comment by davidvossel** [comment]
/hold cancel
/pj-rehearse ack

Even though the jobs are still in progress, i can see that the mce lane conformance tests passed and that the standard aws e2e also passed.

**Pull request merged**
